refactor: Improve search experience and UI and fix an illustration

Joseph-Sameh-0 opened the pull request:
This commit introduces several enhancements to the search functionality and user interface:

- **Search ViewModels:**
    - Increased debounce delay from 500ms to 1000ms for search queries in `SearchViewModel`, `FindByActorViewModel`, and `WorldTourViewModel` to improve performance and reduce unnecessary API calls.
    - Set `isLoading` to `true` when a search query is initiated and `isLoading` to `false` when recent searches are collected in `SearchViewModel`.
    - Set `isLoading` to `true` when a search query is initiated in `FindByActorViewModel`.
    - In `WorldTourViewModel`:
        - Updated `hints` to be a list of `Country` objects.
        - Set `isLoading` to `true` when a search query starts and `false` if no country code or hints are found. - Prioritize search by the first hint if `countryCode` is null but hints are available.

- **Search UI (`NoSearchQueryContent.kt`):**
    - Changed the icon for "Find By Actor" search suggestion.
    - Made the "Recent Searches" title and "Clear All" button visibility dependent on the presence of recent searches using `AnimatedVisibility`.
    - Adjusted padding for `NetworkError`, `PageLoadingPlaceHolder`, and `PlaceholderView`.
    - Updated the subtitle of the placeholder view when there's no search history.

- **WorldTourScreen:**
    - Modified search suggestions to display both country name and code (e.g., "Spain (ES)").

- **Localization:**
    - Added Arabic translation for "Start exploring! Search for your favorite movies, series and shows".
    - Added English string for "Start exploring! Search for your favorite movies, series and shows".

- **Domain Logic (`GetCountryCodeByNameUseCase.kt`):**
    - Ensured the returned country code is uppercase when a direct match with `countryCode` is found.

- **Build Configuration:**
    - Moved `VERSION_NAME` from `Configurations.kt` to `app/build.gradle.kts`.

Joseph-Sameh-0 committed:
refactor: Improve search experience and UI and fix an illustration

This commit introduces several enhancements to the search functionality and user interface:

- **Search ViewModels:**
    - Increased debounce delay from 500ms to 1000ms for search queries in `SearchViewModel`, `FindByActorViewModel`, and `WorldTourViewModel` to improve performance and reduce unnecessary API calls.
    - Set `isLoading` to `true` when a search query is initiated and `isLoading` to `false` when recent searches are collected in `SearchViewModel`.
    - Set `isLoading` to `true` when a search query is initiated in `FindByActorViewModel`.
    - In `WorldTourViewModel`:
        - Updated `hints` to be a list of `Country` objects.
        - Set `isLoading` to `true` when a search query starts and `false` if no country code or hints are found.
        - Prioritize search by the first hint if `countryCode` is null but hints are available.

- **Search UI (`NoSearchQueryContent.kt`):**
    - Changed the icon for "Find By Actor" search suggestion.
    - Made the "Recent Searches" title and "Clear All" button visibility dependent on the presence of recent searches using `AnimatedVisibility`.
    - Adjusted padding for `NetworkError`, `PageLoadingPlaceHolder`, and `PlaceholderView`.
    - Updated the subtitle of the placeholder view when there's no search history.

- **WorldTourScreen:**
    - Modified search suggestions to display both country name and code (e.g., "Spain (ES)").

- **Localization:**
    - Added Arabic translation for "Start exploring! Search for your favorite movies, series and shows".
    - Added English string for "Start exploring! Search for your favorite movies, series and shows".

- **Domain Logic (`GetCountryCodeByNameUseCase.kt`):**
    - Ensured the returned country code is uppercase when a direct match with `countryCode` is found.

- **Build Configuration:**
    - Moved `VERSION_NAME` from `Configurations.kt` to `app/build.gradle.kts`.

Mohamed-Elhanafy commented on app/build.gradle.kts:
```diff
@@ -19,7 +19,7 @@ android {
         targetSdk = Configurations.TARGET_SDK
         versionCode = Configurations.VERSION_CODE
-        versionName = Configurations.VERSION_NAME
+        versionName = "0.1.2"
```
this should be 0.1.3

Mohamed-Elhanafy reviewed (approved):
try to update the relese notes

MustafaIbrahim96 commented:
Please check why those tests failed
WorldTourViewModelTest > onSearchQueryChange should trigger autocomplete and update hints() FAILED
FindByActorViewModelTest > onSearchQueryChange should debounce search calls() FAILED

Before merging it to develop

Joseph-Sameh-0 committed:
fix Debounce timeout in FindByActorViewModelTest

The debounce timeout in `FindByActorViewModelTest` was increased from 600ms to 1100ms. This change ensures that the test accurately reflects the intended debounce behavior, where only the last search query triggers an API call after a 1-second delay.

The release notes in the CD workflow for Firebase App Distribution were also updated to reflect recent improvements in search experience, UI, and an illustration fix.

Joseph-Sameh-0 committed:
fix hint verification in WorldTourViewModelTest

The test for `WorldTourViewModel` was incorrectly asserting a hardcoded list of hints. This commit updates the assertion to use the `mockHints` variable, which reflects the expected output from the `autoCompleteCountryUseCase` mock.

MustafaIbrahim96 commented on app/build.gradle.kts:
```diff
@@ -19,7 +19,7 @@ android {
         targetSdk = Configurations.TARGET_SDK
         versionCode = Configurations.VERSION_CODE
-        versionName = Configurations.VERSION_NAME
+        versionName = "0.1.2"
```
Why do we make the version name hard-coded?

Joseph-Sameh-0 replied:
To prevent build all the modules if the version name changed + it's used only here

MustafaIbrahim96 commented on feature/search/searchUi/src/main/java/com/feature/search/searchUi/screen/findByActor/FindByActorViewModel.kt:
```diff
@@ -57,9 +57,14 @@ class FindByActorViewModel(
         )
         debounceJob?.cancel()
-        if (query.isNotEmpty()) {
+        if (query.isNotBlank()) {
```
Good enhancement, we use isNotBlank to check if the query is not empty.
(For example, when user enters query like this: "    ", that mean query is empty.)

MustafaIbrahim96 reviewed (approved):
great work
